sensor/kubernetes/ping/ping.go

```diff
@@ -0,0 +1,93 @@
+package ping
+
+import (
+	"time"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/centralcaps"
+	"github.com/stackrox/rox/sensor/common/message"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	_ common.SensorComponent = (*pingComponent)(nil)
+
+	p    *pingComponent
+	once sync.Once
+)
+
+// Singleton returns a singleton instance of the ping component.
+func Singleton() common.SensorComponent {
+	once.Do(func() {
+		p = newPingComponent()
+	})
+	return p
+}
+
+func newPingComponent() *pingComponent {
+	return &pingComponent{
+		responsesC:   make(chan *message.ExpiringMessage),
+		stopSignal:   concurrency.NewSignal(),
+		pingInterval: env.PingInterval.DurationSetting(),
+	}
+}
+
+type pingComponent struct {
+	responsesC   chan *message.ExpiringMessage
+	pingInterval time.Duration
+	ticker       *time.Ticker
+	stopSignal   concurrency.Signal
+}
+
+func (p *pingComponent) Notify(_ common.SensorComponentEvent) {}
+
+func (p *pingComponent) Start() error {
+	p.ticker = time.NewTicker(p.pingInterval)
+	go p.run()
+	return nil
+}
+
+func (p *pingComponent) Stop(_ error) {
+	p.ticker.Stop()
+	p.stopSignal.Signal()
+}
+
+func (p *pingComponent) Capabilities() []centralsensor.SensorCapability {
+	return nil
+}
+
+func (p *pingComponent) run() {
+	for {
+		select {
+		case <-p.ticker.C:
+			// Since Start() is called before the initial sync between Central and Sensor has been finalized, we cannot
+			// get away with not initializing nor stopping the ticker.
```

Comment:
Starting in 4.2 we are allowing sensor to run in "offline mode", where components should run in a degraded mode and can choose to buffer or not send information while the connection is broken. 

You can use the `Notify` function to listen to state changes, and start/stop the ticker based on the connection state rather than on `Start` and `Stop` `sensor.Component` lifecycle functions. 

See [cluster metrics](https://github.com/stackrox/stackrox/blob/master/sensor/kubernetes/clustermetrics/cluster_metrics.go#L71) as an example.

`SensorComponentEventCentralReachable` is already being sent since 4.0, so if this is going to patch older releases, you can safely rely on that to start the ticker.